### ISSUE TITLE
`example_proto_coder_test` fails due to absl flags not parsed, and the flags defined seems unnecessary

### ISSUE THREAD

**Issue by Sidong-Wei** [tensorflow/transform#193]
When executing the unit test module `tensorflow_transform/coders/example_proto_coder_test.py`, the following error happens multiple times and almost every sub-test case is failing:
```
======================================================================
ERROR: test_example_proto_coder_picklable (coders.example_proto_coder_test.ExampleProtoCoderTest)
test_example_proto_coder_picklable (coders.example_proto_coder_test.ExampleProtoCoderTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sidong/src/transform/tensorflow_transform/coders/example_proto_coder_test.py", line 72, in setUp
    assert api_implementation.Type() == flags.FLAGS.proto_implementation_type
  File "/usr/local/lib/python3.6/dist-packages/absl/flags/_flagvalues.py", line 491, in __getattr__
    raise _exceptions.UnparsedFlagAccessError(error_message)
absl.flags._exceptions.UnparsedFlagAccessError: Trying to access flag --proto_implementation_type before flags were parsed.

----------------------------------------------------------------------
``` 

I looked into the issue a little bit, and noticed two issues. The first one is that the flag `proto_implementation_type` defined [here](https://github.com/tensorflow/transform/blob/master/tensorflow_transform/coders/example_proto_coder_test.py#L50) is not parsed before being accessed [here](https://github.com/tensorflow/transform/blob/master/tensorflow_transform/coders/example_proto_coder_test.py#L72), and that is the main reason that the error message above appears.
According to Abseil's [documentation](https://abseil.io/docs/python/guides/app), in order for the flags to be parsed, we need to use `app.run()` as the entry point. In other words, the following patch might be needed:
```diff
diff --git a/tensorflow_transform/coders/example_proto_coder_test.py b/tensorflow_transform/coders/example_proto_coder_test.py
index 4a71b0b..e0e4cdf 100644
--- a/tensorflow_transform/coders/example_proto_coder_test.py
+++ b/tensorflow_transform/coders/example_proto_coder_test.py
@@ -24,6 +24,7 @@ import sys
 # GOOGLE-INITIALIZATION
 
 from absl import flags
+from absl import app
 
 # Note that this needs to happen before any non-python imports, so we do it
 # pretty early on.
@@ -199,4 +200,4 @@ class ExampleProtoCoderTest(test_case.TransformTestCase):
 
 
 if __name__ == '__main__':
-  test_case.main()
+  app.run(test_case.main())
```

After I apply the patch and run the test file, the flags could be parsed correctly, but I got the Assertion error, because `api_implementation.Type()` returns `python` while the flag `proto_implementation_type` is defined as `cpp`:
```
(Pdb) l
 68  	class ExampleProtoCoderTest(test_case.TransformTestCase):
 69  	
 70  	  def setUp(self):
 71  	    super(ExampleProtoCoderTest, self).setUp()
 72  	    # Verify that the implementation we requested via the Flag is honoured.
 73  ->	    import pdb; pdb.set_trace()
 74  	    assert api_implementation.Type() == flags.FLAGS.proto_implementation_type
 75  	
 76  	  def assertSerializedProtosEqual(self, a, b):
 77  	    np.testing.assert_equal(_binary_to_example(a), _binary_to_example(b))
 78  	
(Pdb) p api_implementation.Type()
'python'
(Pdb) p flags.FLAGS.proto_implementation_type
'cpp'
```

Basically, the test case failure is due to these two issues. I do not quite understand why we need to check the api implementation here, but if I comment out this [line](https://github.com/tensorflow/transform/blob/master/tensorflow_transform/coders/example_proto_coder_test.py#L72), the entire test module will pass (and the `app.run()` patch above is also not needed). So I just wonder what is the rationale behind this api check and whether we could remove it so that the test case will pass. Thanks.

**Comment by zoyahav** [comment]
What command are you using to run the test?
We do continuously run these tests in virtualenv and it passes, so it could be an environment issue.

**Comment by Sidong-Wei** [comment]
Thanks for the reply. `python3 -m unittest discover -v -p '*_test.py'` is the command I used to run all test cases.
However, if I simply run the single test file by `python3 tensorflow_transform/coders/example_proto_coder_test.py`, I could avoid the "flags not parsed" error, but I still got the assertion error mentioned above.

**Comment by Sidong-Wei** [comment]
Hi @zoyahav, just wonder is there any update on this issue? Thanks.

**Comment by zoyahav** [comment]
No updates, in order to get the test passing either pass it a different flag value (python), or make sure that [fast python protobuf](https://developers.google.com/protocol-buffers/docs/reference/python-generated#cpp_impl) is being used in your environment (this TFT coder works with both, but the cpp variant is more efficient).
As mentioned, we do run this test continuously on several environments, what's the reason that you want to run it locally?

**Comment by Sidong-Wei** [comment]
Thanks for the reply. The reason that I ran the test cases locally is because I am porting TFT to s390x architecture, and this test case failure is the only remaining failure. Since it is also happening on x86 arch in my environment, I am trying to figure out if this failure is crucial in any sense. It's good to know that this flag check is an optional step (since TFT work with both python and cpp variant), so I will close the issue for now, thanks again.
